debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresMetricsIT.java

```diff
@@ -190,6 +191,66 @@ private void assertStreamingMetrics() throws Exception {
         // Assertions.assertThat(mBeanServer.getAttribute(getStreamingMetricsObjectName(), "MonitoredTables")).isEqualTo(new String[] {"public.simple"});
     }
 
+    @Test
+    public void twoRecordsInQueue() throws Exception {
+        final MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        TestHelper.execute(INIT_STATEMENTS, INSERT_STATEMENTS);
+        final int recordCount = 2;
+
+        Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER)
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                .with(PostgresConnectorConfig.MAX_QUEUE_SIZE, 10)
+                .with(PostgresConnectorConfig.MAX_BATCH_SIZE, 1)
+                .with(PostgresConnectorConfig.POLL_INTERVAL_MS, 5000L)
+                .with(PostgresConnectorConfig.MAX_QUEUE_SIZE_IN_BYTES, 10000L);
+        start(PostgresConnector.class, configBuilder.build());
+
+        waitForStreamingToStart();
+        for (int i = 0; i < recordCount - 1; i++) {
+            TestHelper.execute(INSERT_STATEMENTS);
+        }
+        Thread.sleep(1000);
+        Assertions.assertThat(mBeanServer.getAttribute(getStreamingMetricsObjectName(), "CurrentQueueSizeInBytes")).isNotEqualTo(0L);
+        Thread.sleep(500);
+        Assertions.assertThat(mBeanServer.getAttribute(getStreamingMetricsObjectName(), "QueueRemainingCapacity")).isEqualTo(8);
+
+        SourceRecords records = consumeRecordsByTopic(recordCount);
+        Thread.sleep(1000);
+        Assertions.assertThat(mBeanServer.getAttribute(getStreamingMetricsObjectName(), "CurrentQueueSizeInBytes")).isEqualTo(0L);
+        stopConnector();
+    }
+
+    @Test
+    public void oneRecordInQueue() throws Exception {
+        final MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        TestHelper.execute(INIT_STATEMENTS, INSERT_STATEMENTS);
+        final int recordCount = 2;
+
+        Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER)
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                .with(PostgresConnectorConfig.MAX_QUEUE_SIZE, 10)
+                .with(PostgresConnectorConfig.MAX_BATCH_SIZE, 1)
+                .with(PostgresConnectorConfig.POLL_INTERVAL_MS, 5000L)
+                .with(PostgresConnectorConfig.MAX_QUEUE_SIZE_IN_BYTES, 10L);
+        start(PostgresConnector.class, configBuilder.build());
+
+        waitForStreamingToStart();
+        for (int i = 0; i < recordCount - 1; i++) {
+            TestHelper.execute(INSERT_STATEMENTS);
+        }
+        Thread.sleep(1000);
```

Comment:
Test looks good but this is a brittle approach. PLease check `io.debezium.embedded.AbstractConnectorTest.waitForStreamingRunning(String, String, String)` code how to use `Awaitility` lib and use until to wait for CurrentQueueSizeInBytes > 0 and QueueRemainingCapacity == 9
Please follow the same approach in other tests